irohad/ametsuchi/impl/block_serializer.cpp

```diff
@@ -449,70 +458,99 @@ namespace iroha {
+        auto merkle_root_bytes = hex2bytes(merkle_root_str);
+        std::copy(merkle_root_bytes.begin(), merkle_root_bytes.end(),
+                  block.merkle_root.begin());
+      }
 
-      // merkle_root
-      std::string merkle_root_str(doc["merkle_root"].GetString(),
-                                  doc["merkle_root"].GetStringLength());
-      auto merkle_root_bytes = hex2bytes(merkle_root_str);
-      std::copy(merkle_root_bytes.begin(), merkle_root_bytes.end(),
-                block.merkle_root.begin());
-
-      // transactions
-      deserialize(doc, block.transactions);
+      // Deserialize transactions
+      if (not deserialize(doc, block.transactions)) {
+        return nonstd::nullopt;
+      };
 
       return block;
     }
 
     nonstd::optional<model::Transaction> BlockSerializer::deserialize(
         GenericValue<rapidjson::UTF8<char>>::Object& json_tx) {
       model::Transaction tx{};
-      auto json_sigs = json_tx["signatures"].GetArray();
-      for (auto sig_iter = json_sigs.begin(); sig_iter < json_sigs.end();
-           ++sig_iter) {
-        auto json_sig = sig_iter->GetObject();
-        model::Signature signature{};
-
-        std::string sig_pubkey(json_sig["pubkey"].GetString(),
-                               json_sig["pubkey"].GetStringLength());
-        auto sig_pubkey_bytes = hex2bytes(sig_pubkey);
-        std::copy(sig_pubkey_bytes.begin(), sig_pubkey_bytes.end(),
-                  signature.pubkey.begin());
-
-        std::string sig_sign(json_sig["signature"].GetString(),
-                             json_sig["signature"].GetStringLength());
-        auto sig_sign_bytes = hex2bytes(sig_sign);
-        std::copy(sig_sign_bytes.begin(), sig_sign_bytes.end(),
-                  signature.signature.begin());
-
-        tx.signatures.push_back(signature);
+      if (json_tx.HasMember("signatures")) {
+        auto json_sigs = json_tx["signatures"].GetArray();
+        for (auto sig_iter = json_sigs.begin(); sig_iter < json_sigs.end();
+             ++sig_iter) {
+          auto json_sig = sig_iter->GetObject();
+          model::Signature signature{};
+
+          std::string sig_pubkey(json_sig["pubkey"].GetString(),
+                                 json_sig["pubkey"].GetStringLength());
+          auto sig_pubkey_bytes = hex2bytes(sig_pubkey);
+          std::copy(sig_pubkey_bytes.begin(), sig_pubkey_bytes.end(),
+                    signature.pubkey.begin());
+
+          std::string sig_sign(json_sig["signature"].GetString(),
+                               json_sig["signature"].GetStringLength());
+          auto sig_sign_bytes = hex2bytes(sig_sign);
+          std::copy(sig_sign_bytes.begin(), sig_sign_bytes.end(),
+                    signature.signature.begin());
+
+          tx.signatures.push_back(signature);
+        }
+      }
+      if (json_tx.HasMember("created_ts")) {
+        // created_ts
+        tx.created_ts = json_tx["created_ts"].GetUint64();
```

Comment:
Shall we return here false when created_ts is absent? I think this field is necessary for stateless validation, so we should not allow keep it empty